### PR TITLE
Modification due to Labi

### PR DESCRIPTION
**Pull request by EreliaStudio**

### CODE REVIEW

**Comment by EreliaStudio** [comment]
There is a list of documentation error you need to fix before merging this PullRequest :


- Sparkle/src/design_pattern/spk_event_manager.hpp:9
     Compound spk::EventManager is not documented.



- Sparkle/src/design_pattern/spk_event_manager.hpp:12
     Member Callback (typedef) of class spk::EventManager is not documented.



- Sparkle/src/design_pattern/spk_event_manager.hpp:13
     Member Contract (typedef) of class spk::EventManager is not documented.



- Sparkle/src/design_pattern/spk_event_manager.hpp:24
     Member subscribe(const TEventType &p_event, const Callback &p_callback) (function) of class spk::EventManager is not documented.



- Sparkle/src/design_pattern/spk_event_manager.hpp:35
     Member notify(const TEventType &p_event) (function) of class spk::EventManager is not documented.



- Sparkle/src/design_pattern/spk_notifier.hpp:150
     Member Notifier(const Notifier &p_other) (function) of class spk::Notifier is not documented.



- Sparkle/src/design_pattern/spk_notifier.hpp:152
     Member operator=(const Notifier &p_other) (function) of class spk::Notifier is not documented.



- Sparkle/src/design_pattern/spk_singleton.hpp:51
     Member _instance (variable) of class spk::Singleton is not documented.



- Sparkle/src/game_engine/component/spk_camera_component.hpp:118
     Member type() const (function) of class spk::Camera is not documented.



- Sparkle/src/game_engine/component/spk_camera_component.hpp:126
     Member nearPlane() const (function) of class spk::Camera is not documented.



- Sparkle/src/game_engine/component/spk_camera_component.hpp:134
     Member farPlane() const (function) of class spk::Camera is not documented.



- Sparkle/src/game_engine/component/spk_camera_component.hpp:144
     Member fov() const (function) of class spk::Camera is not documented.



- Sparkle/src/game_engine/component/spk_camera_component.hpp:154
     Member aspectRatio() const (function) of class spk::Camera is not documented.



- Sparkle/src/game_engine/component/spk_camera_component.hpp:164
     Member orthographicSize() const (function) of class spk::Camera is not documented.



- Sparkle/src/game_engine/spk_game_object.hpp:77
     Member GameObject(const GameObject &p_other) (function) of class spk::GameObject is not documented.



- Sparkle/src/game_engine/spk_game_object.hpp:78
     Member operator=(const GameObject &p_other) (function) of class spk::GameObject is not documented.



- Sparkle/src/game_engine/spk_mesh.hpp:207
     Member clear() (function) of class spk::Mesh is not documented.



- Sparkle/src/game_engine/spk_mesh.hpp:236
     Member clearPoints() (function) of class spk::Mesh is not documented.



- Sparkle/src/game_engine/spk_mesh.hpp:265
     Member clearUVs() (function) of class spk::Mesh is not documented.



- Sparkle/src/game_engine/spk_mesh.hpp:294
     Member clearNormals() (function) of class spk::Mesh is not documented.



- Sparkle/src/game_engine/spk_mesh.hpp:325
     Member clearIndexes() (function) of class spk::Mesh is not documented.



- Sparkle/src/graphics/texture/spk_texture.hpp:157
     Member Texture(Texture &&p_other) (function) of class spk::Texture is not documented.



- Sparkle/src/graphics/texture/spk_texture.hpp:158
     Member operator=(Texture &&p_other) (function) of class spk::Texture is not documented.



- Sparkle/src/graphics/texture/spk_texture.hpp:157
     Member Texture(Texture &&p_other) (function) of class spk::Texture is not documented.



- Sparkle/src/graphics/texture/spk_texture.hpp:158
     Member operator=(Texture &&p_other) (function) of class spk::Texture is not documented.



- Sparkle/src/graphics/texture/spk_texture.hpp:157
     Member Texture(Texture &&p_other) (function) of class spk::Texture is not documented.



- Sparkle/src/graphics/texture/spk_texture.hpp:158
     Member operator=(Texture &&p_other) (function) of class spk::Texture is not documented.



- Sparkle/src/math/spk_vector3.hpp:428
     Member xy() (function) of interface spk::IVector3 is not documented.



- Sparkle/src/math/spk_vector3.hpp:433
     Member xz() (function) of interface spk::IVector3 is not documented.



- Sparkle/src/math/spk_vector3.hpp:438
     Member yz() (function) of interface spk::IVector3 is not documented.

**Comment by EreliaStudio** [comment]
There is a list of documentation error you need to fix before merging this PullRequest :


- Sparkle/src/design_pattern/spk_event_manager.hpp:9
     Compound spk::EventManager is not documented.



- Sparkle/src/design_pattern/spk_event_manager.hpp:12
     Member Callback (typedef) of class spk::EventManager is not documented.



- Sparkle/src/design_pattern/spk_event_manager.hpp:13
     Member Contract (typedef) of class spk::EventManager is not documented.



- Sparkle/src/design_pattern/spk_event_manager.hpp:24
     Member subscribe(const TEventType &p_event, const Callback &p_callback) (function) of class spk::EventManager is not documented.



- Sparkle/src/design_pattern/spk_event_manager.hpp:35
     Member notify(const TEventType &p_event) (function) of class spk::EventManager is not documented.



- Sparkle/src/design_pattern/spk_notifier.hpp:150
     Member Notifier(const Notifier &p_other) (function) of class spk::Notifier is not documented.



- Sparkle/src/design_pattern/spk_notifier.hpp:152
     Member operator=(const Notifier &p_other) (function) of class spk::Notifier is not documented.



- Sparkle/src/design_pattern/spk_singleton.hpp:51
     Member _instance (variable) of class spk::Singleton is not documented.



- Sparkle/src/game_engine/component/spk_camera_component.hpp:118
     Member type() const (function) of class spk::Camera is not documented.



- Sparkle/src/game_engine/component/spk_camera_component.hpp:126
     Member nearPlane() const (function) of class spk::Camera is not documented.



- Sparkle/src/game_engine/component/spk_camera_component.hpp:134
     Member farPlane() const (function) of class spk::Camera is not documented.



- Sparkle/src/game_engine/component/spk_camera_component.hpp:144
     Member fov() const (function) of class spk::Camera is not documented.



- Sparkle/src/game_engine/component/spk_camera_component.hpp:154
     Member aspectRatio() const (function) of class spk::Camera is not documented.



- Sparkle/src/game_engine/component/spk_camera_component.hpp:164
     Member orthographicSize() const (function) of class spk::Camera is not documented.



- Sparkle/src/game_engine/spk_game_object.hpp:77
     Member GameObject(const GameObject &p_other) (function) of class spk::GameObject is not documented.



- Sparkle/src/game_engine/spk_game_object.hpp:78
     Member operator=(const GameObject &p_other) (function) of class spk::GameObject is not documented.



- Sparkle/src/game_engine/spk_mesh.hpp:207
     Member clear() (function) of class spk::Mesh is not documented.



- Sparkle/src/game_engine/spk_mesh.hpp:236
     Member clearPoints() (function) of class spk::Mesh is not documented.



- Sparkle/src/game_engine/spk_mesh.hpp:265
     Member clearUVs() (function) of class spk::Mesh is not documented.



- Sparkle/src/game_engine/spk_mesh.hpp:294
     Member clearNormals() (function) of class spk::Mesh is not documented.



- Sparkle/src/game_engine/spk_mesh.hpp:325
     Member clearIndexes() (function) of class spk::Mesh is not documented.



- Sparkle/src/graphics/texture/spk_texture.hpp:157
     Member Texture(Texture &&p_other) (function) of class spk::Texture is not documented.



- Sparkle/src/graphics/texture/spk_texture.hpp:158
     Member operator=(Texture &&p_other) (function) of class spk::Texture is not documented.



- Sparkle/src/graphics/texture/spk_texture.hpp:157
     Member Texture(Texture &&p_other) (function) of class spk::Texture is not documented.



- Sparkle/src/graphics/texture/spk_texture.hpp:158
     Member operator=(Texture &&p_other) (function) of class spk::Texture is not documented.



- Sparkle/src/graphics/texture/spk_texture.hpp:157
     Member Texture(Texture &&p_other) (function) of class spk::Texture is not documented.



- Sparkle/src/graphics/texture/spk_texture.hpp:158
     Member operator=(Texture &&p_other) (function) of class spk::Texture is not documented.



- Sparkle/src/math/spk_vector3.hpp:428
     Member xy() (function) of interface spk::IVector3 is not documented.



- Sparkle/src/math/spk_vector3.hpp:433
     Member xz() (function) of interface spk::IVector3 is not documented.



- Sparkle/src/math/spk_vector3.hpp:438
     Member yz() (function) of interface spk::IVector3 is not documented.

**Comment by EreliaStudio** [comment]
There is a list of documentation error you need to fix before merging this PullRequest :


- Sparkle/src/design_pattern/spk_event_manager.hpp:31
     Member Callback (typedef) of class spk::EventManager is not documented.



- Sparkle/src/design_pattern/spk_event_manager.hpp:32
     Member Contract (typedef) of class spk::EventManager is not documented.



- Sparkle/src/design_pattern/spk_notifier.hpp:150
     Member Notifier(const Notifier &p_other) (function) of class spk::Notifier is not documented.



- Sparkle/src/design_pattern/spk_notifier.hpp:152
     Member operator=(const Notifier &p_other) (function) of class spk::Notifier is not documented.



- Sparkle/src/graphics/texture/spk_texture.hpp:157
     Member Texture(Texture &&p_other) (function) of class spk::Texture is not documented.



- Sparkle/src/graphics/texture/spk_texture.hpp:158
     Member operator=(Texture &&p_other) (function) of class spk::Texture is not documented.



- Sparkle/src/graphics/texture/spk_texture.hpp:157
     Member Texture(Texture &&p_other) (function) of class spk::Texture is not documented.



- Sparkle/src/graphics/texture/spk_texture.hpp:158
     Member operator=(Texture &&p_other) (function) of class spk::Texture is not documented.



- Sparkle/src/graphics/texture/spk_texture.hpp:157
     Member Texture(Texture &&p_other) (function) of class spk::Texture is not documented.



- Sparkle/src/graphics/texture/spk_texture.hpp:158
     Member operator=(Texture &&p_other) (function) of class spk::Texture is not documented.

File: Sparkle/src/design_pattern/spk_notifier.cpp [hunk 1]
```diff
@@ -31,6 +39,9 @@ namespace spk
         _deactivated(p_other._deactivated.load())
     {
         p_other._deactivated = true;
+
+        p_other._contracts.clear();
+        p_other._inactiveContracts.clear();
```
Comment (joackimdb) [inline Sparkle/src/design_pattern/spk_notifier.cpp:54]: Moving contracts means that we need to change which Notifier they know.

**Comment by EreliaStudio** [comment]
There is a list of documentation error you need to fix before merging this PullRequest :


- Sparkle/src/data_structure/spk_pool.hpp:41
     Member Allocator (typedef) of class spk::Pool is not documented.



- Sparkle/src/data_structure/spk_pool.hpp:80
     Member editAllocator(const Allocator &p_allocator) (function) of class spk::Pool is not documented.



- Sparkle/src/data_structure/spk_pool.hpp:85
     Member allocate() (function) of class spk::Pool is not documented.



- Sparkle/src/data_structure/spk_pool.hpp:69
     argument 'p_poolSize' of command @param is not found in the argument list of spk::Pool< TType >::Pool(const Allocator &p_allocator=[](){return(new TType());})



- Sparkle/src/data_structure/spk_pool.hpp:74
     The following parameter of spk::Pool::Pool(const Allocator &p_allocator=[](){return(new TType());}) is not documented:
      parameter 'p_allocator'



- Sparkle/src/game_engine/spk_game_object.hpp:185
     Member components() const (function) of class spk::GameObject is not documented.



- Sparkle/src/data_structure/spk_pool.hpp:69
     argument 'p_poolSize' of command @param is not found in the argument list of spk::Pool< TType >::Pool(const Allocator &p_allocator=[](){return(new TType());})



- Sparkle/src/data_structure/spk_pool.hpp:74
     The following parameter of spk::Pool::Pool(const Allocator &p_allocator=[](){return(new TType());}) is not documented:
      parameter 'p_allocator'

File: Sparkle/src/design_pattern/spk_singleton.hpp [hunk 2]
```diff
@@ -43,12 +43,24 @@ namespace spk
...
+          }
 
-		static inline std::unique_ptr<TType> _instance = nullptr;
+          /**
+           * @brief Static unique pointer to the singleton instance.
+           * 
+           * This pointer manages the lifetime of the singleton instance. It ensures that there is only one instance of the
+           * singleton class at any time and provides automatic cleanup.
+           */
+          static inline std::unique_ptr<TType> _instance = nullptr;
```
[the remark this answers is not in the document]
Comment (EreliaStudio) [reply Sparkle/src/design_pattern/spk_singleton.hpp:63]: For automatic desallocation uppon program closure.
If its a TType* and the user forgot to call release, it wont be destroyed, and therefor wont call its destructor. With a unique_ptr, uppon program closure, the destructor will be automaticaly called, except for the case "the user already called release"

**Comment by EreliaStudio** [comment]
There is a list of documentation error you need to fix before merging this PullRequest :


- Sparkle/src/data_structure/spk_pool.hpp:41
     Member Allocator (typedef) of class spk::Pool is not documented.



- Sparkle/src/data_structure/spk_pool.hpp:80
     Member editAllocator(const Allocator &p_allocator) (function) of class spk::Pool is not documented.



- Sparkle/src/data_structure/spk_pool.hpp:85
     Member allocate() (function) of class spk::Pool is not documented.



- Sparkle/src/data_structure/spk_pool.hpp:69
     argument 'p_poolSize' of command @param is not found in the argument list of spk::Pool< TType >::Pool(const Allocator &p_allocator=[](){return(new TType());})



- Sparkle/src/data_structure/spk_pool.hpp:74
     The following parameter of spk::Pool::Pool(const Allocator &p_allocator=[](){return(new TType());}) is not documented:
      parameter 'p_allocator'



- Sparkle/src/game_engine/spk_game_object.hpp:185
     Member components() const (function) of class spk::GameObject is not documented.



- Sparkle/src/widget/spk_widget.hpp:82
     Member _onGeometryChange() (function) of class spk::IWidget is not documented.



- Sparkle/src/widget/spk_widget.hpp:84
     Member _onRender() (function) of class spk::IWidget is not documented.



- Sparkle/src/widget/spk_widget.hpp:86
     Member _onUpdate() (function) of class spk::IWidget is not documented.



- Sparkle/src/data_structure/spk_pool.hpp:69
     argument 'p_poolSize' of command @param is not found in the argument list of spk::Pool< TType >::Pool(const Allocator &p_allocator=[](){return(new TType());})



- Sparkle/src/data_structure/spk_pool.hpp:74
     The following parameter of spk::Pool::Pool(const Allocator &p_allocator=[](){return(new TType());}) is not documented:
      parameter 'p_allocator'

**Pull request merged**